examples/material_gallery/lib/demo/leave_behind_demo.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+enum LeaveBehindDemoAction {
+  reset,
+  horizontalSwipe,
+  leftSwipe,
+  rightSwipe
+}
+
+class LeaveBehindItem {
+  LeaveBehindItem({ this.index, this.name, this.subject, this.body });
+
+  LeaveBehindItem.from(LeaveBehindItem item)
+    : index = item.index, name = item.name, subject = item.subject, body = item.body;
+
+  final int index;
+  final String name;
+  final String subject;
+  final String body;
+}
+
+class LeaveBehindDemo extends StatefulComponent {
+  LeaveBehindDemo({ Key key }) : super(key: key);
+
+  LeaveBehindDemoState createState() => new LeaveBehindDemoState();
+}
+
+class LeaveBehindDemoState extends State<LeaveBehindDemo> {
+  final GlobalKey<ScaffoldState> scaffoldKey = new GlobalKey<ScaffoldState>();
+  DismissDirection dismissDirection = DismissDirection.horizontal;
+  List<LeaveBehindItem> leaveBehindItems;
+
+  void initListItems() {
+    leaveBehindItems = new List.generate(12, (int index) {
```

Comment:
this looks like it could be a static, in which case you wouldn't need the initState